### DISCUSSION TITLE
How I can add border design to `nvim-cmp` and `blink.cmp` `completion and discussion float window` ?

### DISCUSSION THREAD

**Discussion by harshv5094** [LazyVim/LazyVim#4163] [Q&A] [answered]
So I wanted to add borders to nvim-cmp completion and description window.

This is my current code in my LazyVim Config
```lua
{
	"hrsh7th/nvim-cmp",
	dependencies = { "hrsh7th/cmp-emoji" },
	opts = function(_, opts)
		table.insert(opts.sources, { name = "emoji" })
	end,
},
```

I'm adding this code
```lua
{
	"hrsh7th/nvim-cmp",
	dependencies = { "hrsh7th/cmp-emoji" },
	opts = function(_, opts)
                local cmp = require("cmp")
		table.insert(opts.sources, { name = "emoji" })
		table.insert(opts.window, {
                   completion = cmp.config.window.bordered(),
                   documentation = cmp.config.window.bordered(), -- from github repo example
            })
	end,
},
```

Still I'm not getting any border, I don't know why, but the code also turns off cmp suggestion. Does someone have any solution for this?

**Comment by dpetka2001** [comment]
Change your `opts.window` to 
```lua
      opts.window = {
        completion = cmp.config.window.bordered(),
        documentation = cmp.config.window.bordered(), -- from github repo example
      }
```

**Reply by harshv5094** [reply]
I tried it, after reloading nvim-cmp just stops showing suggestion

**Reply by dpetka2001** [reply]
Don't reload, since it doesn't always work correctly. Just restart Neovim. I have it like the code i mentioned and it shows borders correctly.

**Reply by dpetka2001** [reply]
Just 
```lua
{
	"hrsh7th/nvim-cmp",
	dependencies = { "hrsh7th/cmp-emoji" },
	opts = function(_, opts)
        local cmp = require("cmp")
		table.insert(opts.sources, { name = "emoji" })
		opts.window = {
                   completion = cmp.config.window.bordered(),
                   documentation = cmp.config.window.bordered(), -- from github repo example
        }
	end,
},
```

**Reply by harshv5094** [reply, marked as answer]
Hey I read the doc and found the solution, Just needed to do this
```lua
	{
		"hrsh7th/nvim-cmp",
		dependencies = {
			{ "hrsh7th/cmp-emoji" },
		},
		opts = function(_, opts)
			table.insert(opts.sources, {
				{ name = "emoji" },
				{ name = "orgmode" },
			})
			local cmp = require("cmp")
			opts.window = {
				completion = cmp.config.window.bordered({}),
				documentation = cmp.config.window.bordered({}),
			}
			-- Set view to follow cursor while typing
			opts.view = {
				entries = {
					follow_cursor = true,
				},
			}
		end,
	},
```

**Reply by harshv5094** [reply]
It's working now!!! 🥳🥳🥳🥳🥳🥳🥳, just needed to pass table to `bordered`

![image](https://github.com/user-attachments/assets/96294ea1-cd8e-4ee7-9367-8ea5d00ae0f5)

**Reply by dpetka2001** [reply]
It's not actually needed to pass a table into `bordered`. I have it without and it's working just fine. But since you got it working, that's what matters.

**Reply by apaydev** [reply]
I don't know if something changed, but this is not working anymore. Here's my config:
```lua
return {
  {
    "hrsh7th/nvim-cmp",
    opts = function(_, opts)
      local cmp = require("cmp")
      opts.window = {
        completion = cmp.config.window.bordered({}),
        documentation = cmp.config.window.bordered({}),
      }
    end,
  },
}
```
![image](https://github.com/user-attachments/assets/bd6d2e10-b1d8-4fdc-a2ff-c50db585c0bb)

**Reply by PauloHenriqueOliveiradeAlmeida** [reply]
same problem here, I searched in many forums and asked for the chat-GPT, but I couldn't find the solution, did you manage to find the problem?

**Reply by apaydev** [reply]
@PauloHenriqueOliveiradeAlmeida Yeah, on Lazyvim 14.x nvim-cmp was replaced by blink.cmp, as specified in [here](https://www.lazyvim.org/news).

There is a way to still use nvim-cmp, which is specified on the link above. However, I decided to add a config file for blink.

Give me a few hours and I'll share my updated dotfiles here.

**Reply by dpetka2001** [reply]
For blink you can do `opts.completion.menu.border = "rounded"` (or whatever other style you like) and for documentation `opts.completion.documentation.window.border = "rounded"`. Take a look at [blink's docs](https://cmp.saghen.dev/configuration/reference.html#completion-menu) for more info.

**Reply by PauloHenriqueOliveiradeAlmeida** [reply]
With Blink configuration that's work perfectly, thank's for the answer to my question, see my Blink configuration file:

```lua
return {
  {
    "saghen/blink.cmp",
    opts = {
      completion = {
        menu = {
          border = "rounded",
        },
        documentation = {
          window = {
            border = "rounded",
          },
        },
      },
    },
  },
}
```

The beatiful result:
![image](https://github.com/user-attachments/assets/d5eba81d-041d-4661-a779-ab3f0fa921bf)

**Reply by harshv5094** [reply]
I've also switched to `blink.cmp` now @PauloHenriqueOliveiradeAlmeida. So your solution helps me out. Thanks!

**Reply by devnchill** [reply]
thanks @PauloHenriqueOliveiradeAlmeida this works

**Reply by kailiu42** [reply]
As doc of blink.cmp states border defaults to `vim.o.winborder` on nvim 0.11+, we can just set it accordingly.

**Comment by harshv5094** [comment]
I'm changing the title for future visitors
